### PR TITLE
Fix format migration regression

### LINKED ISSUE #5667: Problem to start minio server in distributed mode after upgrade to 2018-03-16T22:52:12Z
I just upgrade minio to latest version.

```
$ minio version
Version: 2018-03-16T22:52:12Z
Release-Tag: RELEASE.2018-03-16T22-52-12Z
Commit-ID: f23944aed7b7980d4400ac8bc022dabd8ac1c273
```

and after upgrade all 4 server, it's problem to start minio server in distributed mode.

when running command on first server,

```minio server http://node1/mnt/minio/cluster-data http://node2/mnt/minio/cluster-data http://node3/mnt/minio/cluster-data http://node4/mnt/minio/cluster-data```

server start and `"Waiting for a minimum of 2 disks to come online"`

when I run same command on other servers, I see error
```
Trace: 1: /q/.q/sources/gopath/src/github.com/minio/minio/cmd/server-main.go:247:cmd.serverMain()
       2: /q/.q/sources/gopath/src/github.com/minio/minio/vendor/github.com/minio/cli/app.go:499:cli.HandleAction()
       3: /q/.q/sources/gopath/src/github.com/minio/minio/vendor/github.com/minio/cli/command.go:214:cli.Command.Run()
       4: /q/.q/sources/gopath/src/github.com/minio/minio/vendor/github.com/minio/cli/app.go:260:cli.(*App).Run()
       5: /q/.q/sources/gopath/src/github.com/minio/minio/cmd/main.go:155:cmd.Main()
       6: /q/.q/sources/gopath/src/github.com/minio/minio/main.go:71:main.main()
[2018-03-18T13:50:14.132962493Z] [ERROR] Initializing object layer failed (http://node3:9000/mnt/minio/cluster-data format error: Disk ID  not found in any disk sets 
[[427bef43-e2bf-4e79-a7bd-00b820ba241e c7524dab-2dcd-4747-8dd5-89a593cd3b8d 3bdea66e-5698-477a-bfe2-7d331fff979d faa851b2-d9a1-4cb1-b369-da674846f6b6]])
```

format.json
```json
{"version":"1","format":"xl","xl":{
"version":"3",
"this":"",
"sets":[
["427bef43-e2bf-4e79-a7bd-00b820ba241e",
"c7524dab-2dcd-4747-8dd5-89a593cd3b8d",
"3bdea66e-5698-477a-bfe2-7d331fff979d",
"faa851b2-d9a1-4cb1-b369-da674846f6b6"]],
"distributionAlgo":"CRCMOD"}}
```

## Possible Solution
Thank for help @y4m4 on slack
Manual setup UUID per server in format.json and minio server start working again.

server1:
```json
{"version":"1","format":"xl","xl":{
"version":"3","this":"427bef43-e2bf-4e79-a7bd-00b820ba241e",
"sets":[["427bef43-e2bf-4e79-a7bd-00b820ba241e","c7524dab-2dcd-4747-8dd5-89a593cd3b8d","3bdea66e-5698-477a-bfe2-7d331fff979d","faa851b2-d9a1-4cb1-b369-da674846f6b6"]],"distributionAlgo":"CRCMOD"}}
```

server2:
```json
{"version":"1","format":"xl","xl":{
"version":"3","this":"c7524dab-2dcd-4747-8dd5-89a593cd3b8d",
"sets":[["427bef43-e2bf-4e79-a7bd-00b820ba241e","c7524dab-2dcd-4747-8dd5-89a593cd3b8d","3bdea66e-5698-477a-bfe2-7d331fff979d","faa851b2-d9a1-4cb1-b369-da674846f6b6"]],"distributionAlgo":"CRCMOD"}}
```
server3:
```json
{"version":"1","format":"xl","xl":{
"version":"3","this":"3bdea66e-5698-477a-bfe2-7d331fff979d",
"sets":[["427bef43-e2bf-4e79-a7bd-00b820ba241e","c7524dab-2dcd-4747-8dd5-89a593cd3b8d","3bdea66e-5698-477a-bfe2-7d331fff979d","faa851b2-d9a1-4cb1-b369-da674846f6b6"]],"distributionAlgo":"CRCMOD"}}
```
server4:
```json
{"version":"1","format":"xl","xl":{
"version":"3","this":"faa851b2-d9a1-4cb1-b369-da674846f6b6",
"sets":[["427bef43-e2bf-4e79-a7bd-00b820ba241e","c7524dab-2dcd-4747-8dd5-89a593cd3b8d","3bdea66e-5698-477a-bfe2-7d331fff979d","faa851b2-d9a1-4cb1-b369-da674846f6b6"]],"distributionAlgo":"CRCMOD"}}
```

## Environment
Linux  4.13.0-37-generic #42~16.04.1-Ubuntu SMP Wed Mar 7 16:03:28 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
current minio version: 2018-03-16T22:52:12Z
old minio version: 1 month old (I dont know exactly)

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Migration regression got introduced in 9083bc152e798026f348357d68623444b098d174
adding more unit tests to catch this scenario, we need to fix this by
re-writing the formats after the migration to 'V3'.

This bug only happens when a user is migrating directly from V1 to V3,
not from V1 to V2 and V2 to V3.

Added additional unit tests to cover these situations as well.

## Motivation and Context
Fixes #5667

## How Has This Been Tested?
Manually and unit tested.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :x: [more...](#b4918bea-7809-4891-9d37-c53629737411) |
<a name="b4918bea-7809-4891-9d37-c53629737411"></a>
### 5668-3ff7921/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT: minikube:31223
ACCESS_KEY:      minio
SECRET_KEY:      ***REDACTED***
ENABLE_HTTPS:    0
SERVER_REGION:   us-east-1
MINT_DATA_DIR:   /mint/data
MINT_MODE:       core

To get logs, run 'docker cp 5f6c228e454f:/mint/log /tmp/mint-logs'
(1/12) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-r6l6kpmlqjiwz0",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 335,
  "error": "RequestError: send request failed\ncaused by: Put http://minikube:31223/aws-sdk-go-test-r6l6kpmlqjiwz0: dial tcp 192.168.122.45:31223: getsockopt: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 12 tests successfully.
```

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :x: [more...](#3ce3e0a3-8551-42d7-af08-82d69c769883) |
| mint-gateway-azure.sh | :x: [more...](#81fb12b1-9218-4725-a10a-ab94a432b3f9) |
<a name="81fb12b1-9218-4725-a10a-ab94a432b3f9"></a>
### 5668-9562c00/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT: minikube:31739
ACCESS_KEY:      minioazure1
SECRET_KEY:      ***REDACTED***
ENABLE_HTTPS:    0
SERVER_REGION:   us-east-1
MINT_DATA_DIR:   /mint/data
MINT_MODE:       core

To get logs, run 'docker cp 383ed9693eeb:/mint/log /tmp/mint-logs'
(1/12) Running aws-sdk-go tests ... done in 1 seconds
(2/12) Running aws-sdk-php tests ... done in 6 minutes and 17 seconds
(3/12) Running aws-sdk-ruby tests ... done in 47 seconds
(4/12) Running awscli tests ... FAILED in 3 minutes and 22 seconds
{
  "name": "awscli",
  "duration": 7401,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (BucketAlreadyOwnedByYou) when calling the CreateBucket operation: Your previous request to create the named bucket succeeded and you already own it."
}

Executed 3 out of 12 tests successfully.
```
<a name="3ce3e0a3-8551-42d7-af08-82d69c769883"></a>
### 5668-9562c00/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT: minikube:30468
ACCESS_KEY:      minio
SECRET_KEY:      ***REDACTED***
ENABLE_HTTPS:    0
SERVER_REGION:   us-east-1
MINT_DATA_DIR:   /mint/data
MINT_MODE:       core

To get logs, run 'docker cp 3238b4063a20:/mint/log /tmp/mint-logs'
(1/12) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-bqfabeha12u0y2",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 400,
  "error": "RequestError: send request failed\ncaused by: Put http://minikube:30468/aws-sdk-go-test-bqfabeha12u0y2: dial tcp 192.168.122.45:30468: getsockopt: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 12 tests successfully.
```

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#2f0ecd78-a5b2-45d2-9562-0d3111753c0c) |
<a name="2f0ecd78-a5b2-45d2-9562-0d3111753c0c"></a>
### 5668-b9cf2c4/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT: minikube:30987
ACCESS_KEY:      minioazure1
SECRET_KEY:      ***REDACTED***
ENABLE_HTTPS:    0
SERVER_REGION:   us-east-1
MINT_DATA_DIR:   /mint/data
MINT_MODE:       core

To get logs, run 'docker cp 84727c120421:/mint/log /tmp/mint-logs'
(1/12) Running aws-sdk-go tests ... done in 1 seconds
(2/12) Running aws-sdk-php tests ... done in 6 minutes and 31 seconds
(3/12) Running aws-sdk-ruby tests ... done in 1 minutes and 10 seconds
(4/12) Running awscli tests ... done in 3 minutes and 59 seconds
(5/12) Running mc tests ... done in 8 minutes and 30 seconds
(6/12) Running minio-dotnet tests ... done in 1 minutes and 21 seconds
(7/12) Running minio-go tests ... done in 1 seconds
(8/12) Running minio-java tests ... done in 44 seconds
(9/12) Running minio-js tests ... FAILED in 50 seconds
{
  "name": "minio-js",
  "function": "makeBucket(bucketName, region)",
  "args": "bucketName:minio-js-test-c7a2f309-dede-46a8-83ee-41f1ded12d6d-region, region:us-east-1",
  "duration": 40201,
  "status": "FAIL",
  "error": "S3Error: We encountered an internal error, please try again. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:56:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:117:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:48:26) at DestroyableTransform.<anonymous> (node_modules/minio/node_modules/through2/node_modules/readable-stream/lib/_stream_transform.js:135:12) at finishMaybe (node_modules/minio/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:371:12) at endWritable (node_modules/minio/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:378:3) at DestroyableTransform.Writable.end (node_modules/minio/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:356:5) at IncomingMessage.onend (_stream_readable.js:511:10) at endReadableNT (_stream_readable.js:974:12) at _combinedTickCallback (internal/process/next_tick.js:80:11) at process._tickCallback (internal/process/next_tick.js:104:9)"
}

Executed 8 out of 12 tests successfully.
```

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

**Review by krishnasrinivas — DISMISSED** [review]
Tested

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

File: cmd/format-xl.go [hunk 1]
```diff
@@ -602,6 +603,44 @@ func initStorageDisks(endpoints EndpointList) ([]StorageAPI, error) {
 	return storageDisks, nil
 }
 
+// formatXLV3ThisEmpty - need to find out if .This filed is still empty.
```
Comment (krisis) [inline cmd/format-xl.go]: typo: s/filed/field

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

**Review by krisis — APPROVED** [review]
LGTM, tested locally by migrating a local single node erasure code backend setup with 4 drives, from version 1 to 3.

**Review by krishnasrinivas — APPROVED** [review]

**Pull request merged**